### PR TITLE
Add better description to the tests

### PR DESCRIPTION
**Pull request by itepifanio**
This PR add a better description to the "tests" section from the README.md, it was based on the [JOSS review feedback](https://github.com/openjournals/joss-reviews/issues/4480#issuecomment-1184651947).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,9 +30,18 @@ Ipyannotator provides a [simple API](https://palaimon.github.io/ipyannotator/nbs
 
 ### Run ipyannotator tests
 
-The tests can be executed using `nbdev_test_nbs` on the terminal. 
+Ipyannotator's tests were written on the Jupyter Notebook, following the literate programming style used on the library, and can be executed by running `nbdev_test_nbs` on the terminal.
```
Comment (ibayer) [inline README.md:33]:
```suggestion
Ipyannotator's tests are directly integrated in the Jupyter Notebooks that make up the libraries source code, this is a consequence of following the literate programming style made possible by the library `nbdev`. Tests can be executed by running `nbdev_test_nbs` on the terminal.
```

File: README.md [hunk 2]
```diff
@@ -30,9 +30,18 @@ Ipyannotator provides a [simple API](https://palaimon.github.io/ipyannotator/nbs
...
 
-Ipyannotator notebooks contains the development dependencies: [pytest](https://docs.pytest.org/en/7.1.x/) and [ipytest](https://github.com/chmp/ipytest). Make sure this dependencies are installed to run the tests.
+[Nbdev](https://nbdev.fast.ai/#A-Motivating-Example) it's the tool that allow Ipyannotator to be written in notebooks. The tool uses cell comments, such as `#exports`, to output the library code, and cells without comments to be executed as tests. Nbdev by itselft doesn't guarantee that the tests are self-contained, that's why Ipyannotator uses [pytest](https://docs.pytest.org/en/7.1.x/) and [ipytest](https://github.com/chmp/ipytest) as dev dependencies. To execute the tests the dev dependencies mentioned need to be installed.
```
Comment (ibayer) [inline README.md:35]: move this paragraph to the bottom of this section, this background information is not need to run the tests

File: README.md [hunk 3]
```diff
@@ -30,9 +30,18 @@ Ipyannotator provides a [simple API](https://palaimon.github.io/ipyannotator/nbs
...
+
+When installing the repository using poetry, all dev dependencies are installed.
```
Comment (ibayer) [inline README.md:37]:
```suggestion
When installing the repository using poetry, all dev dependencies are installed by default.
```

File: README.md [hunk 4]
```diff
@@ -30,9 +30,18 @@ Ipyannotator provides a [simple API](https://palaimon.github.io/ipyannotator/nbs
...
+
+When using pip make sure to install `pytest` and `ipytest`:
```
Comment (ibayer) [inline README.md:39]:
```suggestion
When using pip for installation make sure to install the two dev dependencies `pytest` and `ipytest`, with the versions listed in `pyproject.toml`, manually:
```

**Review by ibayer — CHANGES_REQUESTED** [review]
@itepifanio 
Just a few suggestions for edits.

**Comment by itepifanio** [comment]
@ibayer thanks for your suggestions, my last commit should address them all.

**Comment by ibayer** [comment]
@itepifanio 
Thanks, merging. :)

**Pull request merged**